src/crawlee/storages/_key_value_store.py

```diff
@@ -157,3 +157,15 @@ async def set_value(
             return await self._resource_client.delete_record(key)
 
         return await self._resource_client.set_record(key, value, content_type)
+
+    def get_public_url(self, key: str) -> str:
+        """Get the public URL for the given key.
+
+        Args:
+            key: Key of the record for which URL is required
+        Returns:
+            The public URL for the given key.
+        """
+        name = self.name or self._configuration.default_key_value_store_id
+        storage_dir = self._configuration.storage_dir
+        return f'file://{storage_dir}/storage/key_value_stores/{name}/{key}'
```

Comment:
Issue #514 mentions that we should put the actual logic for determining the public url into the `KeyValueStoreClient` class (file `src/crawlee/memory_storage_client/_key_value_store_client.py`). And call the method from here. Can you please do that?

This will allow the Apify SDK to replace the functionality in a cleaner fashion.

Reply:
Moved the implementation to client